### PR TITLE
One Odra Core to rule them all

### PR DESCRIPTION
**Pull request by kubaplas**
Merged odra-utils, odra-casper-shared and odra-types into odra-core

### CODE REVIEW

File: core/src/host_env.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ impl HostEnv {
         let use_proxy = T::cl_type() != <()>::cl_type() || !call_def.attached_value().is_zero();
         let call_result = backend.call_contract(&address, call_def, use_proxy);
 
-        let mut events_map: BTreeMap<Address, Vec<odra_types::Bytes>> = BTreeMap::new();
+        let mut events_map: BTreeMap<Address, Vec<crate::Bytes>> = BTreeMap::new();
```
Comment (kpob) [inline core/src/host_env.rs]:
```suggestion
        let mut events_map: BTreeMap<Address, Vec<Bytes>> = BTreeMap::new();
```
already imported in L9

File: odra-macros/src/utils/string.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+///
+/// ```
+/// use odra_utils::camel_to_snake;
+///
+/// let camel = "ContractName";
+/// let result = camel_to_snake(camel);
+///
+/// assert_eq!(&result, "contract_name");
+/// ```
+pub fn camel_to_snake<T: ToString>(text: T) -> String {
```
Comment (kpob) [inline odra-macros/src/utils/string.rs:15]: Should be removed and use the function from `odra_core::utils`

File: odra-macros/Cargo.toml [hunk 3]
```diff
@@ -11,11 +11,11 @@ homepage = { workspace = true }
 repository = { workspace = true }
 
 [dependencies]
-odra-utils = { path = "../utils", features = ["std"] }
 proc-macro2 = "1.0.69"
 quote = "1.0.33"
 syn = { version = "2.0.29", features = ["full"] }
 syn_derive = "0.1.8"
+convert_case = { version = "0.5.0" }
```
Comment (kpob) [inline odra-macros/Cargo.toml:18]: Not needed if using odra-core

**Review by kpob — CHANGES_REQUESTED** [review]
#264 introduced some changes that centralize usage of `odra::*` paths. If you sync with the current release/0.8.0 it reduces changes in `odra-macros` crate. You me face a lot of conflicts but you should take those from release and apply changes in `odra_macros::utils::*` modules.

File: odra-macros/Cargo.toml (hunk 3 above)
Comment (kubaplas) [reply odra-macros/Cargo.toml:18]: This crate is not no-std, it's simpler this way

Comment (kpob) [reply odra-macros/Cargo.toml:18]: Ok.

**Review by kpob — APPROVED** [review]

**Pull request merged**